src/reporting.ts

```diff
@@ -56,18 +54,20 @@ export const getNotesInMarkdown = (): string => {
 };
 
 export const generateMarkdownReport = (): void => {
-    const workspaceFolder = vscode.workspace.rootPath;
-    // TODO: What if there's no workspace?
-    if (workspaceFolder) {
-        // TODO: Remove this hardcoded string, it should be a configuration
-        const extensionDirPath = path.join(workspaceFolder, '.vscode', 'code-annotation');
-        const extensionFilePath = path.join(extensionDirPath, 'summary.md');
-        let content = getNotesInMarkdown();
-        fs.writeFileSync(extensionFilePath, content);
-        var openPath = vscode.Uri.file(extensionFilePath);
-        vscode.workspace.openTextDocument(openPath).then(doc => {
-            vscode.window.showTextDocument(doc).then(editor => {
-            });
+    const newFile = vscode.Uri.parse('untitled:summary.md');
+
+    vscode.workspace.openTextDocument(newFile).then(summaryFile => {
+        const edit = new vscode.WorkspaceEdit();
+        let notesSummary = getNotesInMarkdown();
+
+        edit.insert(newFile, new vscode.Position(0, 0), notesSummary);
+
+        return vscode.workspace.applyEdit(edit).then(success => {
+            if (success) {
+                vscode.window.showTextDocument(summaryFile);
```

Comment:
According to the API for showTextDocument, we can force the new document to be brought to focus by passing a preservefocus = false. I think for this functionality this would be great.

```suggestion
                vscode.window.showTextDocument(summaryFile, /*column=*/undefined, /*preserveFocus=*/false);
```

Reply:
I can't reproduce it not being brought to focus when generating the summary, but I'm commiting your suggestion to be sure 😅